pkg/sync/v6/client.go

```diff
@@ -254,3 +259,24 @@ func printDiff(log *logrus.Entry, existing, o *unstructured.Unstructured) bool {
 	}
 	return diffShown
 }
+
+// reconcileProtected check if object is marked with azure reconcile protection annotation
+// this will remove certain objects from being updated post cluster creation
+// access to modify these objects is limited by RBAC
+func (s *sync) reconcileProtected(existing, o *unstructured.Unstructured) bool {
+	if strings.ToLower(o.GetAnnotations()[reconcileProtectKeyAnnotationKey]) == "true" {
```

Comment:
isn't this a logic error?  Isn't the point of reconcileProtectKeyAnnotationKey that it's in the user's control?  We wouldn't ever want to see it in our yaml files?

Reply:
It's an open question. We dont use it now, but I see potential use-cases in the future. But we can add them as we add the use-cases :)